spec/porky_lib/unencrypted/file_service_spec.rb

```diff
@@ -60,100 +61,209 @@ def write_test_file(data)
   end
 
   describe '#write' do
-    it 'write plaintext data to S3' do
-      file_key = file_service.write(plaintext_data, bucket_name)
-      expect(file_key).not_to be_nil
+    it 'raises FileServiceError when file is nil' do
+      expect do
+        file_service.write(nil, bucket_name)
+      end.to raise_exception(PorkyLib::Unencrypted::FileService::FileServiceError)
     end
 
-    it 'write large plaintext data to S3' do
-      file_key = file_service.write(File.read("spec#{File::SEPARATOR}porky_lib#{File::SEPARATOR}data#{File::SEPARATOR}large_plaintext"),
-                                    bucket_name)
-      expect(file_key).not_to be_nil
+    it 'raises FileServiceError when bucket name is nil' do
+      expect do
+        file_service.write(plaintext_data, nil)
+      end.to raise_exception(PorkyLib::Unencrypted::FileService::FileServiceError)
     end
 
-    it 'write plaintext data to S3 with metadata' do
-      metadata = { content_type: 'test/data' }
-      file_key = file_service.write(plaintext_data, bucket_name, metadata: metadata)
+    it 'writes the right content to S3 if file object is used' do
+      allow(Aws::S3::Object).to receive(:new).and_return(aws_s3_object)
+      allow(aws_s3_object).to receive(:upload_file)
+
+      file = write_test_file(plaintext_data)
+      file_service.write(file, bucket_name)
+
+      expect(aws_s3_object).to have_received(:upload_file) do |tempfile_path|
+        expect(File.read(tempfile_path)).to eq(plaintext_data)
+      end
+    end
```

Comment:
I know this works for any files, but sometimes plaintext files are simpler. I'd maybe add a test for an image or other type of file?

Reply:
Yeah that can be added, to test that function, the null byte tests are supposed to _simulate_ that, but its obviously not an image